File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

stack = []
n = int(input())

for _ in range(n):
    s = input().split()

    if s[0] == "push":
        stack.append(s[1])

    elif s[0] == "pop":
        if not stack:
            print(-1)
        else:
            print(stack.pop())
            
    elif s[0] == "size":
        print(len(stack))

    elif s[0] == "empty":
        if not stack:
            print(1)
        else:
            print(0)

    else:
        if not stack:
            print(-1)
        else:
            print(stack[-1])

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int stack[10001];
	int topIndex = -1; //현재 위치 표시기

	int n;
	cin >> n;

	while (n--) {
		string cmd;
		cin >> cmd;

		if (cmd == "push") {
			int x;
			cin >> x;
			topIndex++;
			stack[topIndex] = x; //스택에 추가
		}

		else if (cmd == "pop") {
			if (topIndex == -1) cout << -1 << "\n";
			else cout << stack[topIndex--] << "\n"; //출력후 감소
		}

		else if (cmd == "size") {
			cout << topIndex + 1 << "\n"; //0번 인덱스부터 시작하므로 +1
		}

		else if (cmd == "empty") {
			if (topIndex == -1) cout << 1 << "\n";
			else cout << 0 << "\n";
		}

		else if (cmd == "top") {
			if (topIndex == -1) cout << -1 << "\n";
			else cout << stack[topIndex] << "\n";
		}
	}

	return 0;
}